calculate_trigger.py

```diff
@@ -52,9 +74,47 @@ def search_for_fault_location(
         goldenrun_tb_info.at[idtbinfo, "assembler"],
         goldenrun_tb_info.at[idtbinfo, "id"],
     )
+    return [idx, ins]
+
+
+def search_for_fault_location(
+    filter_lists,
+    trigger_position,
+    fault_address,
+    trigger_occurrences,
+    fault_lifespan,
+    goldenrun_tb_exec,
+    goldenrun_tb_info,
+):
+    logger.info(f"Search trigger to fault INSN at 0x{fault_address:08x}")
+    [idx, ins] = find_fault(
+        fault_address, goldenrun_tb_exec, goldenrun_tb_info, trigger_occurrences
+    )
+    if idx < 0:
+        return [-1, trigger_occurrences, fault_lifespan]
     trigger_not_in_same_tb = 0
+    lifespan_diff = trigger_position + fault_lifespan
     trigger_position = trigger_position * (-1)
     while trigger_position != 0:
+        if idx < 0:
+            if fault_lifespan > 0:
+                if lifespan_diff < 0:
+                    logger.warning(
+                        "Lifespan is to shoort to take effect for {} at {} with "
```

Comment:
Please use f-strings and format the output a little, [for example](https://github.com/Fraunhofer-AISEC/archie/pull/38/files#diff-9501e43f51561a116076aad8f2500b2b3fc659267d4a9e36ffba44d3c14f8101R89).

Reply:
Also a typo: short